IaC/variables.tf

```diff
@@ -0,0 +1,39 @@
+variable "db_username" {
```

Comment:
Same as with `outputs`, you need two `variables.tf` files :
* One that lives within the API module's directory
* One for the root module, that contains inputs only for the root module

Plus if you also refactor them based on a "prefix" variable like I suggested in the comment below, you will have way fewer variables to maintain.

Basically, I think all variables here except DB username and password need to be moved to a separate `variables.tf` file in the module specific folder (that you still need to create, comment is in `lambda_module_template.tf`)